packages/synapse-react-client/src/components/EntityCitation/EntityCitation.test.tsx

```diff
@@ -0,0 +1,279 @@
+import { act, render, screen, waitFor } from '@testing-library/react'
+import EntityCitation from './EntityCitation'
+import { useGetEntityBundle } from '@/synapse-queries/entity/useEntityBundle'
+import { useGetDOIAssociation } from '@/synapse-queries/doi/useDOI'
+import { getUseQueryMock } from '@/testutils/ReactQueryMockUtils'
+import {
+  DoiAssociation,
+  EntityBundle,
+  EntityType,
+  ObjectType,
+} from '@sage-bionetworks/synapse-types'
+import {
+  DoiObjectType,
+  SynapseClientError,
+} from '@sage-bionetworks/synapse-client'
+import { createWrapper } from '@/testutils/TestingLibraryUtils'
+import { vi } from 'vitest'
+import userEvent from '@testing-library/user-event'
+
+vi.mock('@/synapse-queries/entity/useEntityBundle', () => {
+  return {
+    useGetEntityBundle: vi.fn(),
+  }
+})
+
+vi.mock('@/synapse-queries/doi/useDOI', () => {
+  return {
+    useGetDOIAssociation: vi.fn(),
+  }
+})
+
+const mockUseGetEntityBundle = vi.mocked(useGetEntityBundle)
+const mockUseGetDOIAssociation = vi.mocked(useGetDOIAssociation)
+
+const {
+  mock: useGetEntityBundleMockImpl,
+  setSuccess: setMockUseGetEntityBundleSuccess,
+} = getUseQueryMock<Partial<EntityBundle>, SynapseClientError>()
+
+const { mock: useGetEntityDOIMockImpl, setSuccess: setEntityDOISuccess } =
+  getUseQueryMock<DoiAssociation | null, SynapseClientError>()
+
+const { mock: useGetProjectDOIMockImpl, setSuccess: setProjectDOISuccess } =
+  getUseQueryMock<DoiAssociation | null, SynapseClientError>()
+
+const doiEntitySuccess: DoiAssociation = {
+  objectId: 'syn61841662',
```

Comment:
It would be much more clear if we replaced these IDs with a variable describing what they represent

const mockEntityWithUnversionedDoiId = 'syn61841662'
```suggestion

const doiEntitySuccess: DoiAssociation = {
  objectId: mockEntityWithUnversionedDoiId,
```